### DISCUSSION TITLE
Predicted vs actual plot

### DISCUSSION THREAD

**Discussion by adgfo** [kogalur/randomForestSRC#308] [Q&A]
Hi, thank you for the RFSRC package.
How can I generate a plot of predicted vs. actual values, including a fitting line?
Thank you!

**Comment by ishwaran** [comment]
Here's an example for regression:

```
o <- rfsrc(mpg~.,mtcars)
plot(o$yvar, o$predicted.oob, ylab="OOB predicted value", xlab="Actual value")
lines(lowess(o$yvar, o$predicted.oob),col=4)
abline(0,1)
```

**Reply by adgfo** [reply]
Thanks!

**Comment by vfincke** [comment]
Hi, 
I am trying to predict survival with your great package but are currently struggling to understand the predicted vs predicted.oob results. I used the above mentioned plot() function to plot OOB predicted value and predicted value and they seem to do have opposite tendencies. Can someone please help me understand this? 

train <- sample(1:nrow(varimp4), round(nrow(varimp4) * 0.60))

varimp.out <- rfsrc(Surv(time, status) ~ ., data = varimp4[train,], ntree = 5000, 
                    nsplit = 1, nodesize = 4, importance = TRUE, splitrule = "logrank")

> print(varimp.out)
                         Sample size: 47
                    Number of deaths: 25
                     Number of trees: 5000
           Forest terminal node size: 4
       Average no. of terminal nodes: 6.1556
No. of variables tried at each split: 7
              Total no. of variables: 42
       Resampling used to grow trees: swor
    Resample size used to grow trees: 30
                            Analysis: RSF
                              Family: surv
                      Splitting rule: logrank *random*
       Number of random split points: 1
                          (OOB) CRPS: 0.24036743
   (OOB) Requested performance error: 0.8452237

 
plot(varimp.out$yvar[,1], varimp.out$predicted, 
     ylab="predicted value", xlab="Actual value(survival in days)", pch = as.numeric(pch_vector),
     main = "varimp.out"
)

![Bildschirmfoto 2023-02-17 um 15 02 38](https://user-images.githubusercontent.com/125280490/219676750-e17f7ff1-8418-4273-80f7-c171f988e9b9.png)

**Comment by ishwaran** [comment]
Since I can't reproduce your example, let me show you how to compare OOB versus Inbag predicted values using the PBC data set.

Try the following code:

```
data(pbc, package = "randomForestSRC")
o <- rfsrc(Surv(days, status) ~ ., pbc)

par(mfrow=c(2,2))
plot(o$predicted.oob, o$yvar[,1], xlab="Time", ylab="OOB mortality")
plot(o$predicted, o$yvar[,1], xlab="Time", ylab="mortality")
plot(o$predicted, o$predicted.oob, xlab="mortality", ylab="OOB mortality")
abline(0,1)

```

**Reply by vfincke** [reply]
Thank you very much for the help!
I am able to reconstruct the plot with the pic data, but for my data, the OOB predicted value, does not have the same tendency as, the Inbag values.. Do you have any explanation, what could lead to results like this? 
I can provide you with my data set, if that would help. At this point I do not know how to proceed further... 
Thanks again! 
[comparison_inbagOOB_whole.pdf](https://github.com/kogalur/randomForestSRC/files/10785366/comparison_inbagOOB_whole.pdf)
[Plot_IB_OOB.pdf](https://github.com/kogalur/randomForestSRC/files/10785367/Plot_IB_OOB.pdf)

**Comment by ishwaran** [comment]
Please see the following vignette: https://www.randomforestsrc.org/articles/forestWgt.html

Basically the OOB estimator is cross-validated and is used for inference on the training data.  The in-bag estimator would overfit in some settings and is not recommended.

**Reply by vfincke** [reply]
Thank you very much for the reply! But I still do not understand, why the OOB mortality is so different from the predicted value. Why does the OOB value increase over time? This does not make much sense als the mortality should be higher in the beginning. 
Or does this mean, that my Data is not suitable for predicting survival? Or should I just ignore this and go on with the inbox values for prediction? 
Thank you very much for the help!

**Comment by ishwaran** [comment]
I think somethings wrong with your data.  Your OOB error rate is really high (~0.845), which is much higher than even 0.5, which is a bad sign.

**Reply by vfincke** [reply]
But the OOB requested performance error is equal to the c-index and for a c-index, the value is pretty good right? Can you tell me how I can interpret this? Or does the c-index not apply here? 

get.cindex(varimp.out$yvar[,1], varimp.out$yvar[,2], varimp.out$predicted.oob)
[1] 0.8452237

**Reply by ishwaran** [reply]
We report 1 minus the C statistic as the error rate. Small error rates are the goal.

**Reply by vfincke** [reply]
Then how is it possible that the requested performance error and the c-index is the same? 
Thank you for the help!

**Comment by ishwaran** [comment]
We define the C-index to be 1-C
